src/init.c

```diff
@@ -648,7 +649,7 @@ static void jl_resolve_sysimg_location(JL_IMAGE_SEARCH rel)
         jl_options.julia_bindir = abspath(jl_options.julia_bindir, 0);
     free(free_path);
     free_path = NULL;
-    if (jl_options.image_file) {
+    if (jl_options.image_file && rel != JL_IMAGE_IN_MEMORY) {
```

Comment:
I think you still need to process this field, as julia expects it to hold a valid value in various places (for Distributed, precompilation, and other similar optimizations).